File: Data Structures & Algorithms/subsets/submission-1.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {

        List<List<Integer>> res   = new ArrayList<>();
        List<Integer>       inner = new ArrayList<>();
        // Initialize results list [[]]
        res.add(inner);

        // Keep track of elements in nums array that have been processed
        for ( int i = 0; i < nums.length; i++ ) {
            
            // Add every new element from nums list to the inner lists of result
            int numOfInnerLists = res.size();
            for ( int k = 0; k < numOfInnerLists; k++ ) {
                // Get a copy of the sublist
                List sub = new ArrayList<>(res.get(k));
                // Add new element to the end of the list
                sub.add(nums[i]);
                // Add new list containing the nums[i] element
                res.add(sub);
            }

        }
        return res;
    }
}

File: Data Structures & Algorithms/find-the-town-judge/submission-1.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        
        if len(trust) == 1:
            return trust[0][1]

        res = -1
        for i in range(len(trust) - 1):
            p1 = trust[i][1]
            p2 = trust[i+1][1]
            res = p1

            if p1 != p2:
                return -1

        return res
